htdocs/accountancy/class/bookkeeping.class.php

```diff
@@ -2707,10 +2706,11 @@ public function closeFiscalPeriod($fiscal_period_id, $new_fiscal_period_id, $sep
 				$sql .= ' AND aa.pcg_type IN (' . $this->db->sanitize(implode(',', $pcg_type_filter), 1) . ')';
 				$sql .= " AND DATE(t.doc_date) >= '" . $this->db->idate($fiscal_period->date_start) . "'";
 				$sql .= " AND DATE(t.doc_date) <= '" . $this->db->idate($fiscal_period->date_end) . "'";
-				$sql .= ' GROUP BY t.numero_compte, t.label_compte, aa.pcg_type';
+				$sql .= ' GROUP BY t.numero_compte, aa.pcg_type';
 				if ($separate_auxiliary_account) {
-					$sql .= ' ,t.subledger_account, t.subledger_label';
+					$sql .= " , NULLIF(t.subledger_account, '')";
```

Comment:
Adding a function call into a group by is strange. DoesYour fix work if you replace this with
`$sql .= ", subledger_account";`

Reply:
Hi @eldy

It does not work if you remove the NULLIF function. We still get duplicates.
Here are some screenshots with and without:

![SQLDIFFWITHOUTISNULLINGROUP](https://github.com/user-attachments/assets/6ec259e5-3957-4823-b3f2-00fc599e36be)
![SQLDIFFWITHISNULLINGROUP](https://github.com/user-attachments/assets/9fa51abd-5d72-4685-92d9-a8b8991c7133)